Runtime/Tracking/Collision/Active/ActiveCollisionsContainer.cs

```diff
@@ -196,15 +224,71 @@ protected virtual CollisionNotifier.EventData CloneEventData(CollisionNotifier.E
         }
 
         /// <summary>
-        /// Determines if the current collision is valid.
+        /// Determines if the current containing <see cref="Transform"/> in the collision is valid.
         /// </summary>
-        /// <param name="collisionData">The collision to check.</param>
+        /// <param name="containingTransform">The <see cref="Transform"/> to check.</param>
         /// <returns>The validity result of the collision.</returns>
-        protected virtual bool IsValidCollision(CollisionNotifier.EventData collisionData)
+        protected virtual bool IsValidCollision(Transform containingTransform)
         {
-            Transform containingTransform = collisionData.ColliderData.GetContainingTransform();
             return containingTransform != null && CollisionValidity.Accepts(containingTransform.gameObject);
         }
 
+        /// <summary>
+        /// Adds the given data to the <see cref="Elements"/> collection if it is not already present.
+        /// </summary>
+        /// <param name="data">The data to add.</param>
+        protected virtual void AddUniqueElement(CollisionNotifier.EventData data)

```

Comment:
Perhaps this whole method should be removed and `Elements` should become `HashSet<CollisionNotifier.EventData>` instead? Better perf that way.

Reply:
The `Elements` are required to be ordered as they are emitted in the event for processing elsewhere.

Making it a HashSet would mean something then would need to turn it back into a List for sorting down the line.

Reply:
`SortedSet` then!

Reply:
I'd say we defer that change for another time as the List is processed in multiple places already meaning this change would get much larger.

Plus right now the Contains check is only done on Start and Stop collision so it's not firing every frame, so the perf hit of the Contains check isn't massive right now.

Reply:
I don't think the public API would change much but 🤷‍♂ fine as is.

Reply:
Things like `AddRange` and accessing with `[index]` don't work with a `SortedSet` which are used elsewhere